### ISSUE TITLE
Mixed `type` imports does not trigger `import/no-duplicates`

### ISSUE THREAD

**Issue by musjj** [import-js/eslint-plugin-import#3035]
This correctly triggers `import/no-duplicates`:

```typescript
import { useEffect } from "react";
import { useState } from "react";
```

But this doesn't:

```typescript
import { type ComponentProps, useEffect } from "react";
import { useState } from "react";
```

The rule should merge the imports above into:

```typescript
import { type ComponentProps, useEffect, useState } from "react";
```

**Comment by ljharb** [comment]
Note this depends on the configuration, because it could also potentially be autofixed into
```js
import type { ComponentProps } from "react";
import { useState, useEffect } from "react";
```

**Comment by KAMRONBEK** [comment]
Confirmed on current `main` (`import/no-duplicates`, default options):

```ts
import { type ComponentProps, useEffect } from 'react';
import { useState } from 'react';
```

is not flagged, while the same pair without the inline `type` is. Root cause is in `getImportMap`: under `prefer-inline: false`, any declaration containing an inline `type` specifier is bucketed into `namedTypesImported`, and duplicates are only checked within a bucket — so a mixed `{ type X, y }` declaration never meets the plain `{ z }` import in `imported`. (It already works under `prefer-inline: true`, because that branch is gated on `!preferInline` and the mixed decl falls through to `map.imported`.)

On @ljharb's point that the autofix direction depends on config — I don't think this needs a new decision, the option already picks the shape:

- `prefer-inline: true` → collapse into one statement with inline types: `import { type ComponentProps, useEffect, useState } from 'react';`. The existing fixer already produces exactly this once detection fires — verified locally.
- `prefer-inline: false` (default) → keep the type specifier as a separate top-level `import type { ComponentProps }` and only merge the value specifiers, which matches the plugin's current default of keeping `import type {}` apart from value imports (and stays consistent with the #2470 handling of inline-type-only declarations).

The mechanical part is trickier than it looks, though. I prototyped the obvious routing tweak (send a decl with any value specifier to the value bucket, keep pure inline-type-only decls where they are). It fixes this in both modes with the correct autofix output and no #2470 regression — but it breaks the existing test `import {AValue, type x, BValue} from './foo'; import {type y} from './foo'` (105→103 passing), because a mixed declaration also has to duplicate a *pure* inline-type declaration of the same source, and simple bucket-routing can't express "duplicate of two different buckets at once." That needs cross-bucket detection.

#2716 already implements essentially this (a dedicated inline-type bucket merged against both the value and type buckets at `Program:exit`, with prefer-inline-aware fixers), which is why I'm raising this here rather than sending a competing patch — it, and the adjacent #3194, both currently show as conflicting against `main` and touch the same routing code.

Before I put up anything: would you prefer this folded into a rebased #2716, or a fresh PR on current `main` that reuses that cross-bucket merge approach? Happy to take on the rebase if that's useful.
